### PR TITLE
feat: Add slack notification support for reports

### PR DESCRIPTION
**Pull request by AdityaHegde**
- [x] Add slack user and channel inputs to create.
- [x] Fix slack notification. (https://github.com/rilldata/rill/pull/6005)
- [x] Add slack user and channel to metadata.
- [x] Add slack user and channel inputs to edit.
- [x] Get feedback on design, the form is too long. Perhaps we can use the new multi entry input used in user invite? (Switched to MultiInput along with using superforms)

### CODE REVIEW

File: web-common/src/features/scheduled-reports/utils.ts [hunk 1]
```diff
@@ -17,3 +67,93 @@ export function getDashboardNameFromReport(
...
+  return mappedEntries as {
+    [KEY in K]: string;
+  }[];
+}
+
+function mapAndAddEmptyEntryV2(entries: string[] | undefined) {
+  const finalEntries = entries ? [...entries] : [];
+  finalEntries.push("");
+  return finalEntries;
+}
```
Comment (ericpgreen2) [inline web-common/src/features/scheduled-reports/utils.ts:159]: Do we need to add an empty entry? It'd be simpler if we didn't.

File: web-common/src/features/scheduled-reports/utils.ts [hunk 2]
```diff
@@ -17,3 +67,93 @@ export function getDashboardNameFromReport(
     null
   );
 }
+
+export function extractNotification(
```
Comment (ericpgreen2) [inline web-common/src/features/scheduled-reports/utils.ts:71]: "extract" seems like the wrong verb here

Comment (ericpgreen2) [inline web-common/src/features/scheduled-reports/utils.ts:71]: I only see this function called once, it doesn't use the `userEmail` parameter, and it only uses `isEdit=true`, so we could remove those two parameters? Or we could consolidate and use `extractNotificationV2`.

File: web-common/src/features/scheduled-reports/utils.ts [hunk 3]
```diff
@@ -1,5 +1,55 @@
...
+    timeZone: reportSpec?.refreshSchedule?.timeZone ?? getLocalIANA(),
+    exportFormat: reportSpec
+      ? (reportSpec?.exportFormat ?? V1ExportFormat.EXPORT_FORMAT_UNSPECIFIED)
+      : V1ExportFormat.EXPORT_FORMAT_CSV,
+    exportLimit: reportSpec
+      ? reportSpec.exportLimit === "0"
+        ? ""
+        : reportSpec.exportLimit
+      : "",
+    ...extractNotificationV2(reportSpec?.notifiers, userEmail, !!reportSpec),
```
Comment (ericpgreen2) [inline web-common/src/features/scheduled-reports/utils.ts:48]: `getNotificationValues` could be a better name

File: web-common/src/features/scheduled-reports/ScheduledReportDialog.svelte [hunk 4]
```diff
@@ -52,124 +47,101 @@
...
-            getRuntimeServiceGetResourceQueryKey($runtime.instanceId, {
-              "name.name": reportName,
-              "name.kind": ResourceKind.Report,
-            }),
-          );
-        }
+        },
+      });
 
+      if (reportSpec) {
```
Comment (ericpgreen2) [inline web-common/src/features/scheduled-reports/ScheduledReportDialog.svelte:102]: Nit: it looks like an explanatory variable `const isEditForm = !!reportSpec` could help both here and in line 118

**Review by ericpgreen2 — CHANGES_REQUESTED** [review]
Design (cc @jkhwu) :
1. For the recipients input, can we make the label "Email recipients"? Else "recipients" and "users" look too similar.
    image
2. Can we have separate tags for "Email sent" and "Slack message sent"? In this case, there was no email sent, only a Slack message. In other cases, it's possible for both an email and a Slack message to be sent.
    image

File: web-common/src/features/scheduled-reports/utils.ts (hunk 2 above)
Comment (AdityaHegde) [reply web-common/src/features/scheduled-reports/utils.ts:71]: Moving this back to alerts. It is till using the old forms lib. We can merge them once we move it to superforms.

File: web-common/src/features/scheduled-reports/utils.ts (hunk 1 above)
Comment (AdityaHegde) [reply web-common/src/features/scheduled-reports/utils.ts:159]: The empty entry is what is edited. It is mainly to support submit without needing to convert the currently being edited entry to a pill explicitly.

File: web-common/src/features/scheduled-reports/utils.ts (hunk 2 above)
Comment (AdityaHegde) [reply web-common/src/features/scheduled-reports/utils.ts:71]: We extract the slack and email related fields from then notifiers array.

File: web-common/src/features/scheduled-reports/utils.ts [hunk 5]
```diff
@@ -17,3 +67,93 @@ export function getDashboardNameFromReport(
     null
   );
 }
+
+export function extractNotification(
+  notifiers: V1Notifier[] | undefined,
```
[the remark this answers is not in the document]
Comment (AdityaHegde) [reply web-common/src/features/scheduled-reports/utils.ts:72]: Moved back to alerts with no changes.

File: web-common/src/features/scheduled-reports/utils.ts [hunk 6]
```diff
@@ -17,3 +67,93 @@ export function getDashboardNameFromReport(
...
+    slackChannels,
+    slackUsers,
+
+    enableEmailNotification: isEdit ? !!emailNotifier : true,
+    emailRecipients,
+  };
+}
+
+export function extractNotificationV2(
+  notifiers: V1Notifier[] | undefined,
```
[the remark this answers is not in the document]
Comment (AdityaHegde) [reply web-common/src/features/scheduled-reports/utils.ts:112]: This is a single function that handles both new and edit use case.

**Comment by AdityaHegde** [comment]
1. Done.
2. A generic `Report sent` is better IMO. If one of them fails to send we should get an error and show that instead.

File: web-common/src/features/scheduled-reports/ScheduledReportDialog.svelte [hunk 7]
```diff
@@ -52,124 +47,102 @@
...
-              "name.name": reportName,
-              "name.kind": ResourceKind.Report,
-            }),
-          );
-        }
+        },
+      });
 
+      const isEdit = !!reportSpec;
+      if (isEdit) {
```
Comment (ericpgreen2) [inline web-common/src/features/scheduled-reports/ScheduledReportDialog.svelte:103]: For clarity, this `isEdit` variable could be used in more places throughout this file – wherever there's a ternary based on `reportSpec`

**Review by ericpgreen2 — APPROVED** [review]

**Pull request merged**